modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -1983,6 +1983,57 @@ TEST_P(Test_ONNX_nets, Googlenet)
     expectNoFallbacksFromIE(net);
 }
 
+TEST_P(Test_ONNX_nets, ResNet50)
+{
+    const String model =  findDataFile("dnn/ascend-models/resnet50-v1-12.onnx", true);
+
+    Net net = readNetFromONNX(model);
+    ASSERT_FALSE(net.empty());
+
+    net.setPreferableBackend(backend);
+    net.setPreferableTarget(target);
+
+    Mat inp = imread(_tf("../grace_hopper_227.png"));
+    Mat ref = blobFromNPY(_tf("../caffe_alexnet_prob.npy"));
+    checkBackend(&inp, &ref);
+
+    net.setInput(blobFromImage(inp, 1.0f, Size(224, 224), Scalar(), false));
+    // ASSERT_FALSE(net.empty()); // it is empty
+    TickMeter tm;
+    int repeat = 10;
+    std::vector<double> times;
+    tm.start();
+    Mat out = net.forward();
+    tm.stop();
+    double firstrun = tm.getTimeMilli();
+    std::cout << "firstrun: " << firstrun << std::endl;
+    tm.reset();
+    for (int i = 0; i < repeat; i++)
+    {
+        tm.start();
+        net.forward();
+        tm.stop();
+        times.push_back(tm.getTimeMilli());
+        tm.reset();
+    }
+    double min_time = times[0], max_time = times[0];
+    double total_time = 0;
+    for (auto t : times)
+    {
+        total_time += t;
+        if (min_time > t)
+            min_time = t;
+        if (max_time < t)
+            max_time = t;
+    }
+    std::cout << cv::format("min = %f, max = %f, avg = %f, first run = %f\n", min_time, max_time, total_time/repeat, firstrun);
+
+    Net::finalize();
```

Comment:
@asmorkalov I need to register this API to finalize CANN resources before the program exits. Do you know how to call it after all gtests finish and before the program exits?

Reply:
You can take a look on OpenCL internals: Device and Context. OpenCL context is thread-local object. It's stored in TLS and destroyed automatically with TLS entry. See https://github.com/opencv/opencv/blob/4.x/modules/core/src/ocl.cpp#L1069

Reply:
It is not quite the same situation with OpenCL. `Net::finalize()` calls `ge::aclgrphBuildFinalize()` from the CANN library, and this API needs to be called before a process ends and can only be called once in a process.

Maybe we can address this problem by spawning a child process to build graph and call `ge::aclgrphBuildFinalize()` there. Do we have examples in the opencv source code using multiple processes (something like fork?)? I did a simple search but could not find one.

Reply:
Problem is solved. We do not need `Net::finalizeDevice()` anymore since we compile the CANN model in a child process using fork() and shared_memory. Initialization and finalizing of Graph API are done in the child process.